### ISSUE TITLE
How can I get rid of the "config file template has changed" warning?

### ISSUE THREAD

**Issue by offbyone** [twpayne/chezmoi#2378]
## What exactly are you trying to do?

I have edited my chezmoi.toml template, but my local config is just fine and I really don't want to regenerate it. How do I stop chezmoi from warning me constantly:

```console
❯ chezmoi diff
chezmoi: warning: config file template has changed, run chezmoi init to regenerate config file
```

## What have you tried so far?

I tried to `touch ~/.config/chezmoi/chezmoi.toml` to no avail.

## Where else have you checked for solutions?

* [X] I have read [chezmoi's user guide](https://chezmoi.io/user-guide/), and not found the answer.
* [X] I have searched [chezmoi's reference guide](https://chezmoi.io/reference/), and not found the answer.
* [X] Other, please give details.

I looked for issues here referencing it as well.

## Output of any commands you've tried with `--verbose` flag

```console
$ chezmoi --verbose diff
chezmoi: warning: config file template has changed, run chezmoi init to regenerate config file
```

## Output of `chezmoi doctor`

<details>

```console
$ chezmoi doctor
RESULT   CHECK                MESSAGE
ok       version              v2.23.0, commit 7481b57f40c89a777e1f13be3e1d418a8b4f5a16, built at 2022-09-19T00:15:13Z, built by Homebrew
ok       latest-version       v2.23.0
ok       os-arch              darwin/arm64
ok       uname                Darwin Chriss-MacBook-Pro.local 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000 arm64
ok       go-version           go1.19.1 (gc)
ok       executable           /opt/homebrew/bin/chezmoi
ok       upgrade-method       brew-upgrade
ok       config-file          ~/.config/chezmoi/chezmoi.toml, last modified 2022-09-22T08:07:58-07:00
ok       source-dir           ~/.local/share/chezmoi is a git working tree (clean)
ok       suspicious-entries   no suspicious entries
ok       working-tree         ~/.local/share/chezmoi is a git working tree (clean)
ok       dest-dir             ~ is a directory
ok       umask                022
ok       cd-command           found /opt/homebrew/bin/fish
ok       cd-args              /opt/homebrew/bin/fish
info     diff-command         not set
ok       edit-command         found /opt/homebrew/bin/emacsclient
ok       edit-args            /opt/homebrew/bin/emacsclient
ok       git-command          found /usr/bin/git, version 2.37.0
ok       merge-command        found /usr/bin/vimdiff
ok       shell-command        found /opt/homebrew/bin/fish
ok       shell-args           /opt/homebrew/bin/fish
info     age-command          age not found in $PATH
info     gpg-command          gpg not found in $PATH
info     pinentry-command     not set
info     1password-command    op not found in $PATH
info     bitwarden-command    bw not found in $PATH
info     gopass-command       gopass not found in $PATH
info     keepassxc-command    keepassxc-cli not found in $PATH
info     keepassxc-db         not set
info     keeper-command       keeper not found in $PATH
info     lastpass-command     lpass not found in $PATH
info     pass-command         pass not found in $PATH
info     passhole-command     ph not found in $PATH
info     vault-command        vault not found in $PATH
info     secret-command       not set
```

</details>

**Comment by twpayne** [comment]
The quick and dirty work-around here is:
1. Move your config file somewhere safe.
2. Run `chezmoi init`.
3. Move your config file back.

**Comment by offbyone** [comment]
I've done this before; it's ... not my favorite. Is there no way to do it in a less hacky way? :D

**Comment by twpayne** [comment]
What sort of solution would you like?

One possibility would be to make it possible to disable the warning in chezmoi's config file, something like:

```toml
[warnings]
    configFileTemplateHasChanged = false # default is true
```

How does this sound?

**Comment by offbyone** [comment]
I think I would want to understand what the warning is for in the first place. To me, I would expect that once there _is_ a configuration file, there's no further need to bring the template into the story at all; I've configured chezmoi, and if I want to amend it, I'll explicitly update the config, rather than worrying about the template. I assume you built that warning for a reason, though; would you be willing to explain why?

**Comment by twpayne** [comment]
tl;dr This warning tells you when your config file not up to date with your dotfile templates.

More detail:

I added the warning as I was changing the variables used in my templates and my config file template, which meant that my templates no longer worked with old config files.

My [current config file template](https://github.com/twpayne/dotfiles/blob/63207b2c41cdc8cda18e02186006f4dfc7e0315b/home/.chezmoi.toml.tmpl) uses a bunch of boolean feature flags that are set on a per-machine basis. These boolean feature flags are used in various script and dotfile templates ([example](https://github.com/twpayne/dotfiles/blob/63207b2c41cdc8cda18e02186006f4dfc7e0315b/home/private_dot_config/git/config.tmpl#L95-L100)). When I add a new boolean feature flag and use it in a template I'll get a "map has no entry for key" error on machines with an old config file that does not define the boolean feature flag. This warning alerts me that my config file might not be up to date with respect to my templates.

**Comment by offbyone** [comment]
That's interesting; you have many fewer data values than I do; I configure a lot more in there, and regenerating those has been a pain. Interesting... I wonder if I should refactor.

**Comment by offbyone** [comment]
Okay, having had some time to think about this, I think having a way to suppress these would be nice, since I will nearly always have configuration in mine that is purely host-local and that I don't want to regenerate without care and consideration.
